File: python_Programmers/1단계/예산.py
# 내 풀이
def solution(d, budget):
    cnt = 0
    d.sort()
    for i in d:
        if budget >= i:
            budget -=i
            cnt +=1
        else:
            break
    return cnt

# 모범 풀이
def solution2(d, budget):
    answer = 0
    d = sorted(d)
    for needs in d :
        budget -= needs
        if budget < 0 :
            break
        answer = answer + 1

    return answer

d = [2,2,3,3]
budget = 10
print(solution(d, budget))
print(solution2(d, budget))

File: Java_Programmers/1단계/src/예산.java
import java.util.Arrays;

public class 예산 {
    public int solution(int[] d, int budget) {
        int answer = 0;
        Arrays.sort(d);
        for (int n : d) {
            if(budget < n)
                break;
            budget -= n;
            answer++;
        }
        return answer;
    }

    public static void main(String[] args) {
        예산 sol = new 예산();
        int[] d = {1,3,2,5,4};
        int budget = 9;
        System.out.println(sol.solution(d, budget));
    }
}
